eclair-core/src/main/scala/fr/acinq/eclair/io/Peer.scala

```diff
@@ -347,31 +348,25 @@ object Peer {
 
   // @formatter:on
 
-
-  def generateKey(nodeParams: NodeParams, keyPath: Seq[Long]): PrivateKey = DeterministicWallet.derivePrivateKey(nodeParams.extendedPrivateKey, keyPath).privateKey
-
   def makeChannelParams(nodeParams: NodeParams, defaultFinalScriptPubKey: BinaryData, isFunder: Boolean, fundingSatoshis: Long): LocalParams = {
-    // all secrets are generated from the main seed
-    // TODO: check this
-    val keyIndex = secureRandom.nextInt(1000).toLong
+    // TODO: entropy for channelNumber is too low
+    val channelNumber = secureRandom.nextInt(1000000).toLong
```

Comment:
The entropy is too low and this opens to a vulnerability where an attacker opens a channel with us, then probes us with `open` messages until we choose the same `accept.revocationBasepoint`.

Why don't we choose an e.g. 256 bits random number and split it into 4 longs to create key paths?